redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/termination/action/RedbeamsTerminationActions.java

```diff
@@ -113,18 +113,22 @@ protected void prepareExecution(DeregisterDatabaseServerSuccess payload, Map<Obj
             }
 
             @Override
-            protected Selectable createRequest(RedbeamsContext context) {
+            protected void doExecute(RedbeamsContext context, DeregisterDatabaseServerSuccess payload, Map<Object, Object> variables) throws Exception {
                 // Delete the DB stack here instead of deregistration so that we can keep track of its status
                 // through the termination
-                metricService.incrementMetricCounter(MetricType.DB_TERMINATION_FINISHED, Optional.of(context.getDBStack()));
-                dbStackService.delete(context.getDBStack().getId());
-
-                return new RedbeamsEvent(REDBEAMS_TERMINATION_FINISHED_EVENT.name(), 0L);
+                DBStack dbStack = context.getDBStack();
```

Comment:
I would make an optional from the DBStack right here and use them in the following two statements.